lib/working_hours/config.rb

```diff
@@ -120,14 +146,15 @@ def compile_time time
         time
       end
 
-      def validate_working_hours! week
-        if week.empty?
-          raise InvalidConfiguration.new "No working hours given"
+      def validate_hours! dates, type:
+        case type
+        when :working_hours
+          validate_working_hours! dates
+        when :holiday_hours
+          validate_holiday_hours! dates
```

Comment:
Thanks for the refactoring, this is looking a bit convoluted though to call a single method with two parameters just to then branch on this parameter. I think you should keep calling `validate_working_hours!` and `validate_holiday_hours!` from `precompiled`, and from these two methods, call `validate_hours!` after you verified the keys (so where the duplicated loop was before)

Reply:
Fixed